File: extras/haskell/src/Demands.hs
{--
  Control demands

  Copyright(C) 2021 Simon D.Levy

  MIT License
--}

module Demands where

data Demands = Demands { throttle :: Double
                       , roll :: Double  
                       , pitch :: Double  
                       , yaw :: Double  
                     } deriving (Show)

initialDemands :: Demands

initialDemands = Demands 0 0 0 0


addDemands :: Demands -> Demands -> Demands

addDemands d1 d2 = Demands ((throttle d1) + (throttle d2))
                           ((roll d1) + (roll d2))
                           ((pitch d1) + (pitch d2))
                           ((yaw d1) + (yaw d2))
  

File: webots/controllers/controller/demands.py
#!/usr/bin/python3

import numpy as np
import matplotlib.pyplot as plt

data = np.loadtxt('tmp.csv', delimiter=',', skiprows=3)

plt.subplot(3, 1, 1)
plt.plot(data[:,0])
plt.legend(['throttle'])

plt.subplot(3, 1, 2)
plt.plot(data[:,1])
plt.plot(data[:,2])
plt.legend(['roll', 'pitch'])

plt.subplot(3, 1, 3)
plt.plot(data[:,3])
plt.legend(['yaw'])

plt.show()
